src/databricks/sql/client.py

```diff
@@ -479,9 +550,20 @@ def _prepare_native_parameters(
         """
 
         stmt = stmt
-        params = named_parameters_to_tsparkparams(params)  # type: ignore
 
-        return stmt, params
+        if isinstance(params, dict):
+            dbsql_params = self._prepare_dbsql_params_from_dict(params)
+        if isinstance(params, (list, tuple)):
+            dbsql_params = self._prepare_dbsql_params_from_list(params)
+
+        output = []
+
+        for p in dbsql_params:
```

Comment:
Same comment as elsewhere, prefer list comprehensions for the extremely simple list building.